### DISCUSSION TITLE
Prevent translating text in response

### DISCUSSION THREAD

**Discussion by obones** [toverainc/willow#107] [Q&A] [answered]
Hello,

I'm not sure if it's a willow or willow-inference-server issue, but I find that my French sentences spoken to willow always come back as English sentences in the HTTP response from the inference server (I'm using the default best effort one for now)
I find it rather strange to have automated translation, especially because they display quite a few errors:

1. opposite of what I said: Bonjour becomes Bye
2. not that good: rez de chaussée becomes right footed
3. no translation is done: Au revoir is returned as is.

Did I miss a configuration somewhere? Or is it something that I should be able to control with my own inference server?

**Comment by kristiankielhofner** [comment, marked as answer]
We need to re-work certain aspects of WIS with regard to multi-language support. It's my top priority for WIS and I should be able to get it done today.

For the time being you need to add ```?detect_language=True``` to the WIS server URI.

Your French language should come back but be aware - we're still selecting the "real font" for the display and once we settle on one we will make sure it has full Unicode character support. For characters like "é" the command will work but the display will show some gibberish at that character position as the current font does not have full Unicode character set support.

**Reply by obones** [reply]
No urgency here, I'm just reporting things as I go along.
I added the parameter, and now there is a valid "fr" language in the returned JSON.
Sadly, that JSON does not go all the way to openHAB, only the text portion does, and I'm not sure what can be done about that...

**Reply by kristiankielhofner** [reply]
This is a limitation of the openHAB API that I have raised with them.

From what I can gather, Willow is capable beyond what was previously thought possible for integration to these systems and Willow's ability to dynamically detect and transcribe any of 30+ languages has been previously unconsidered by these platforms. Hopefully they will all implement support for this.
